tests/std/tests/P2321R2_views_zip/test.cpp

```diff
@@ -0,0 +1,838 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <ranges>
+#include <span>
+#include <tuple>
+
+#include <range_algorithm_support.hpp>
+
+template <class... RangeTypes>
+concept CanViewZip = requires(RangeTypes&&... range) {
+    std::views::zip(std::forward<RangeTypes>(range)...);
+};
+
+template <class RangeType>
+using AllView = std::views::all_t<RangeType>;
+
+template <class ElementType>
+concept CanTestElementType =
+    (std::copy_constructible<ElementType> && std::equality_comparable<ElementType>); // Required for iter_swap test
+
+template <class Type1, std::size_t Type1Size, class Type2, std::size_t Type2Size, class Type3, std::size_t Type3Size>
+    requires(CanTestElementType<Type1>&& CanTestElementType<Type2>&& CanTestElementType<Type3>)
+class three_element_test_container {
+private:
+    std::array<Type1, Type1Size> type_one_array;
+    std::array<Type2, Type2Size> type_two_array;
+    std::array<Type3, Type3Size> type_three_array;
+
+public:
+    using element_tuple_type       = std::tuple<Type1, Type2, Type3>;
+    using const_element_tuple_type = std::tuple<const Type1, const Type2, const Type3>;
+
+    using reference_tuple_type              = std::tuple<Type1&, Type2&, Type3&>;
+    using const_reference_tuple_type        = std::tuple<const Type1&, const Type2&, const Type3&>;
+    using rvalue_reference_tuple_type       = std::tuple<Type1&&, Type2&&, Type3&&>;
+    using const_rvalue_reference_tuple_type = std::tuple<const Type1&&, const Type2&&, const Type3&&>;
+
+    static constexpr std::size_t smallest_array_size = (std::min)(Type1Size, (std::min)(Type2Size, Type3Size));
+
+    constexpr three_element_test_container(const std::array<Type1, Type1Size> type_one_init_arr,
+        const std::array<Type2, Type2Size> type_two_init_arr, const std::array<Type3, Type3Size> type_three_init_arr)
+        : type_one_array(type_one_init_arr), type_two_array(type_two_init_arr), type_three_array(type_three_init_arr) {}
+
+    template <std::size_t ElementIndex>
+    constexpr auto get_element_span() {
+        STATIC_ASSERT(ElementIndex < 3);
+
+        if constexpr (ElementIndex == 0)
+            return std::span<Type1, Type1Size>{type_one_array};
+
+        else if constexpr (ElementIndex == 1)
+            return std::span<Type2, Type2Size>{type_two_array};
+
+        else
+            return std::span<Type3, Type3Size>{type_three_array};

```

Comment:
Code style:
 * Always block statement for `if`
 * No newlines for chained `if`